sql/moz-fx-data-shared-prod/udf_js/snake_case_columns/udf.sql

```diff
@@ -0,0 +1,69 @@
+CREATE OR REPLACE FUNCTION udf_js.snake_case_columns(input ARRAY<STRING>)
+RETURNS ARRAY<STRING> DETERMINISTIC
+LANGUAGE js
+AS
+  """
+const REV_WORD_BOUND_PAT = new RegExp(
+  "\\b" + // standard word boundary
+  "|(?<=[a-z][A-Z])(?=\\d*[A-Z])" + // A7Aa -> A7|Aa boundary
+  "|(?<=[a-z][A-Z])(?=\\d*[a-z])" + // a7Aa -> a7|Aa boundary
+  "|(?<=[A-Z])(?=\\d*[a-z])"     // a7A -> a7|A boundary
+);
+
+/**
+ * Convert a name to snake case.
+ * 
+ * The specific implementation here uses regular expressions in order to be compatible across languages.
+ * See https://github.com/acmiyaguchi/test-casing
+ */
+function format(input) {
+  const subbed = input.split('').reverse().join('').replaceAll(".", "_").replace(/[^A-Za-z0-9_]/g, " ");
```

Comment:
This doesn't seem exactly equivalent to the [associated ingestion code](https://github.com/mozilla/gcp-ingestion/blob/dad29698271e543018eddbb3b771ad7942bf4ce5/ingestion-core/src/main/java/com/mozilla/telemetry/ingestion/core/util/SnakeCase.java#L25).  For example, it isn't converting underscores into spaces.

I think this would be more equivalent:
```suggestion
  const subbed = input.split('').reverse().join('').replace(/[^\w]|_/g, " ");
```

Or was there a specific reason for having different logic than the original in this case?

Reply:
I'm now guessing you changed the code due to issues with escaping the backslash in the original regular expression, but if you wrap the JavaScript code in a raw string like I suggested in another comment then the code suggestion I made above should work fine.

Reply:
Ah, yes. That seems to have fixed it